### ISSUE TITLE
Bug: Server crashes by player who wearing UmaSuit on Curios

### ISSUE THREAD

**Issue by cot6** [0999312/umapyoi#46]
### Describe the bug

If the player connect to the server then the game server crashes due to stack overflow.

### Steps to reproduce

Wear a suit and reconnect to the server.

### Expected behavior

The player wears that suit.

### Screenshots

![umapyoi_umasuitcurios_canequip](https://github.com/user-attachments/assets/1910b38c-51a5-4609-b059-fbac4eb69d42)

### Versions

- OS: Windows 11
- Launcher: Modrinth
- Java version: `zulu21.36.17-ca-jre21.0.4-win_x64\bin\javaw.exe`
- Minecraft version: 1.21.1
- Umapyoi version: 31d1bc7da245ff09ccc42b87c29a91358ca670b3

**Comment by schindlershadow** [comment]
I also get this crash
```
OS: linux
Launcher: neoforge 21.1.42
Java version: openjdk 21.0.4
Minecraft version: 1.21.1
Umapyoi version: umapyoi-1.0.2-1.21
```

I crashed 3 times when connecting and get 3 different crash reports:
1 https://pastebin.com/raw/aSJYeU3U
2 https://pastebin.com/raw/dfCF152h
3 https://pastebin.com/raw/9X02S8GE

**Comment by 0999312** [comment]
Fixed in 1.0.3
